### PR TITLE
Added support for Dameng tablespace

### PR DESCRIPTION
**Pull request by Stkai**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.dameng/src/org/jkiss/dbeaver/ext/dameng/model/DamengTablespace.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    public Long getUsedSize() {
+        return usedSize;
+    }
+
+    static class FileCache extends JDBCObjectCache<DamengTablespace, DamengDataFile> {
+
+        @Override
+        protected JDBCStatement prepareObjectsStatement(JDBCSession session, DamengTablespace damengTablespace) throws SQLException {
+            JDBCPreparedStatement dbStat = session.prepareStatement("SELECT df.* " +
+                    "FROM V$TABLESPACE AS ts, V$DATAFILE AS df WHERE ts.ID\n" +
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.dameng/src/org/jkiss/dbeaver/ext/dameng/model/DamengTablespace.java]:
```
"FROM V$TABLESPACE AS ts, V$DATAFILE AS df" +
"\nWHERE ts.ID = df.GROUP_ID AND ts.NAME = ?");
```

I think that conditions should be together in one row.

**Comment by LonwoLonwo** [comment]
Hello @Stkai 

Thanks for your contribution. We will look at it.

**Comment by LonwoLonwo** [comment]
Please add description for your PR.

**Comment by LonwoLonwo** [comment]
Also, I would be very appreciative if you add a [translation](https://github.com/dbeaver/dbeaver/wiki/Localization) to the plugins/org.jkiss.dbeaver.ext.dameng/OSGI-INF/l10n/bundle.properties
(But it is not necessary)

**Comment by Stkai** [comment]
Added support for view Dameng tablespace info,datafile info

**Review by LonwoLonwo — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.dameng.ui/plugin.xml [hunk 2]
```diff
@@ -25,5 +25,17 @@
         <configurator class="org.jkiss.dbeaver.ext.dameng.ui.config.DamengSequenceConfigurator">
             <objectType name="org.jkiss.dbeaver.ext.generic.model.GenericSequence"/>
         </configurator>
+
+        <editor id="source.ddl"
+                class="org.jkiss.dbeaver.ui.editors.sql.SQLSourceViewer"
+                label="%editor.source.ddl.name"
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.dameng.ui/plugin.xml:31]: editor.source.ddl.name and editor.source.ddl.description should have values in the plugins/org.jkiss.dbeaver.ext.dameng.ui/OSGI-INF/l10n/bundle.properties

And yes, you need to create OSGI-INF/l10n/bundle.properties first. Or change %editor.source.ddl.name to the value here.

File: plugins/org.jkiss.dbeaver.ext.dameng.ui/OSGI-INF/l10n/bundle.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+editor.source.ddl.name=DDL
+editor.source.ddl.description=DDL
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.dameng.ui/OSGI-INF/l10n/bundle.properties:2]: And also you need to add OSGI to the plugins/org.jkiss.dbeaver.ext.dameng.ui/build.properties

**Review by LonwoLonwo — APPROVED** [review]

**Comment by LonwoLonwo** [comment]
Thanks. Now, we are waiting for one more contributor. And for the QA Team after that. We have a big release now, so you must be patient because it will take some time.

**Comment by uslss** [comment]
verified

**Pull request merged**